core/quantity/src/Formatter/FormatEnums.ts

```diff
@@ -154,6 +169,25 @@ export function scientificTypeToString(scientificType: ScientificType): string {
   return (scientificType === ScientificType.Normalized) ? "Normalized" : "ZeroNormalized";
 }
 
+/**  @beta   */
+export function parseRatioType(ratioType: string, formatName: string): RatioType {
+  const normalizedValue = ratioType.toLowerCase();
+  for (const key in RatioType) {
```

Comment:
do we really need the for loop and the first if statement?
wouldnt this suffice?
```
const enumValue = RatioType[key as keyof typeof RatioType];
if (enumValue.toLowerCase() === normalizedValue) {
  return enumValue as RatioType;
}
```

Reply:
If we do that, eslint complains because that could include other unwanted properties from the prototype.
This rule here: https://eslint.org/docs/latest/rules/guard-for-in

Reply:
I checked other methods where we do the same thing. Turns out we do not have a lot of string enums which we parse from strings. One other example is SchemaItemType, here: https://github.com/iTwin/itwinjs-core/blob/master/core/ecschema-metadata/src/ECObjects.ts#L162
That uses a switch case method which I'd argue is worse.

Reply:
I'm marking the conversation resolved as it's the only thing preventing the merge. We can revisit this later if you want to follow up @aruniverse